assert/assertions_test.go

```diff
@@ -402,6 +407,72 @@ func TestEqualExportedValues(t *testing.T) {
 	            	+   Exported: (int) 2,
 	            	    notExported: (interface {}) <nil>`,
 		},
+		{
+			value1: S7{
+				Nested{[2]int{1, 2}, [2]int{1, 2}},
+				Nested{[2]int{1, 2}, [2]int{3, 4}},
+			},
+			value2: S7{
+				Nested{[3]int{1, 2, 3}, [2]int{3, 4}},
+				Nested{[2]int{5, 6}, [2]int{7, 8}},
+			},
+			expectedEqual: false,
+			expectedFail: `
+	            	Diff:
+	            	--- Expected
+	            	+++ Actual
+	            	@@ -2,5 +2,6 @@
+	            	  Exported: (assert.Nested) {
+	            	-  Exported: ([2]int) (len=2) {
+	            	+  Exported: ([3]int) (len=3) {
+	            	    (int) 1,
+	            	-   (int) 2
+	            	+   (int) 2,
+	            	+   (int) 3
+	            	   },`,
+		},
+		{
+			value1: S7{
+				Nested{[2]int{1, 2}, [2]int{1, 2}},
+				Nested{[2]int{1, 2}, [2]int{3, 4}},
+			},
+			value2: S7{
+				Nested{[2]int{1, 3}, [2]int{3, 4}},
+				Nested{[2]int{5, 6}, [2]int{7, 8}},
+			},
+			expectedEqual: false,
+			expectedFail: `
+	            	Diff:
+	            	--- Expected
+	            	+++ Actual
+	            	@@ -4,3 +4,3 @@
+	            	    (int) 1,
+	            	-   (int) 2
+	            	+   (int) 3
+	            	   },`,
+		},
+		{
+			value1: S7{
+				Nested{[2]interface{}{nil, 2}, [2]int{1, 2}},
+				Nested{[2]int{1, 2}, [2]int{3, 4}},
+			},
+			value2: S7{
+				Nested{[2]interface{}{nil, 2}, [2]int{3, 4}},
```

Comment:
As `nil` is a special value which means *anything matches*, it would be better to use a concrete value here. Or better: add more test cases to validate that anything matches, not just that `nil` matches *anything*.